packages/react-core/src/hooks/useDropZone.ts

```diff
@@ -85,18 +86,27 @@ export default function useDropZone({
     event.preventDefault();
     event.stopPropagation();
     setDragState('inactive');
-    const files = Array.from(event.dataTransfer.files);
-    const { acceptedFiles, rejectedFiles } = filterAllowedFiles<File>(
-      files,
-      acceptedFileTypes
-    );
+
+    const { files, items } = event.dataTransfer;
 
     if (isFunction(_onDrop)) {
       _onDrop(event);
     }
-    if (isFunction(onDropComplete)) {
-      onDropComplete({ acceptedFiles, rejectedFiles });
-    }
+
+    const completeDrop = (files: File[]) => {
+      const { acceptedFiles, rejectedFiles } = filterAllowedFiles<File>(
+        files,
+        acceptedFileTypes
+      );
+      if (isFunction(onDropComplete)) {
+        onDropComplete({ acceptedFiles, rejectedFiles });
+      }
+      return;
+    };
+
+    !items
+      ? completeDrop(Array.from(files))
+      : processDroppedItems(Array.from(items)).then(completeDrop);
```

Comment:
Ternaries that aren't returning anything read strangely imo

Reply:
Do you think it would read better as an if else?

Reply:
Updated to if else